doc/acl/egress_outer_dscp_change_table.md

```diff
@@ -0,0 +1,505 @@
+            "SRC_IP": "2.2.2.2/32"
+        },
+		"OVERLAY_SET_DSCP_TESTV6|RULE0": {
+            "PRIORITY": "999",
+            "DSCP_ACTION": "41",
+			"SRC_IPV6": "2777::0/64",
+            "DST_IPV6": "2788::0/64"
+        },
+        "OVERLAY_SET_DSCP_TESTV6|RULE1": {
+            "PRIORITY": "998",
+            "DSCP_ACTION": "42",
+			"SRC_IPV6": "2666::0/64",
+            "DST_IPV6": "2555::0/64"
+        }
+    }
+}
+
+```
+
+
+#### CLI
+##### ACL table create CLI 
+
+The existing CLI is extended to support the new table types.
+```
+config acl add table -s <stage> -p <ports> <table_name> <table_type>
+```
+_table_type_ needs to be passed as _"UNDERLAY_SET_DSCPV6"_ or _"UNDERLAY_SET_DSCP"_  to create tables of new types.
+```
+Example : config acl add table -s ingress -p Ethernet0 DATAACL UNDERLAY_SET_DSCP
+Example : config acl add table -s ingress -p Ethernet0 DATAACL UNDERLAY_SET_DSCPV6
+
+```
+
+
+### Warmboot and Fastboot Design Impact  
+
+There is no impact on warmboot or fastboot.
+
+
+### Testing Requirements/Design  
+
+#### Unit Test cases 
+
+- Verify ACL Capability in STATE_DB on supported platforms.
+    - The new field `supported_UnderlaySetDSCPV6` must be true.
+- Verify ACL Capability in STATE_DB on non-supported platforms.
+    - The new field `supported_UnderlaySetDSCPV6` must be false.
+- Verify ACL tables creation and deletion
+    - The UNDERLAY_SET_DSCP and UNDERLAY_SET_DSCPV6 tables can be created and deleted togther and separately.
+- Verify that both UNDERLAY_SET_DSCP/V6 can be bound to same ports as well as separate ports.
+    - The EGR_SET_DSCP table is only created once and is bound to the superset of the ports for both UNDERLAY_SET_DSCP/V6 and removed when all referencing tables are removed.
+- Verify that for multiple tables, different order of creation and removal should not have adverse affect.
+    - The EGR_SET_DSCP table is only created once and is bound to the superset of the ports for both UNDERLAY_SET_DSCP/V6 and removed when all referencing tables are removed.
+- Verify a single EGR_SET_DSP entry is created per a unique DSCP action value in the same table.
+    - If multiple rules share the same SET_DSCP action value only one entry should be created. Also EGR_SET_DSP rule should be created with first entry and removed when last rule is removed.
+- Verify a single EGR_SET_DSP entry is created per a unique DSCP action value accorss different tables.
+    - If multiple rules share the same SET_DSCP action value only one entry should be created. Also EGR_SET_DSP rule should be created with first entry and removed when last rule is removed.
+- Verfiy metadata allocation and deallocation
+    - freed metadata values should be available for next allocation.
+- Verfiy metadata allcoation failure.
+    - when metadata is exhausted, this should result in failure.
+
+
+#### System Test cases 
+
+  New sonic-mgmt(PTF) tests *on supported platforms*
+  * UNDERLAY_SET_DSCP/V6 creation failure test.
```

Comment:
Please also consider the resource competition of this new table and the PFCWD table. It's because we are using EGRESS ACL table to drain the buffer on some platform.
There also Egress ACL table on Mellanox platform in dualtor deployment. We should make sure there is no conflict.

Reply:
These tables are intended for T1 where Overlay Ecmp is being employed. Do we have a PFCWD table for egress on T1.
We can have a paltform specific test which can setup both PFCWD and this table setup on the Egress.